src/aliceVision/version.hpp

```diff
@@ -19,3 +21,40 @@
                              ALICEVISION_TO_STRING(ALICEVISION_VERSION_MINOR) "." \
                              ALICEVISION_TO_STRING(ALICEVISION_VERSION_REVISION)
 
+namespace aliceVision {
+
+class Version
+{
+public:
+    Version(const Vec3 & v) : _v(v)
+    {
+    }
+    
+    Version(int major, int minor, int micro) : _v({major, minor, micro})
+    {
+
+    }
+
+    bool operator<(const Version& other) const
+    {
+        for (Vec3::Index i = 0; i < 3; i++)
+        {
+            if (_v[i] < other._v[i])
+            {
+                return true;
+            }
+
+            if (_v[i] > other._v[i])
+            {
+                return false;
+            }
+        }    
+
+        return false;
+    }
+
+private:
+    Vec3 _v;
```

Comment:
```suggestion
    Vec3i _v;
```

Reply:
It's obvious that integers are better suited for this task.
However, i choose doubles as Json I/O has always loaded/saved doubles.